src/iceberg/manifest_reader_internal.cc

```diff
@@ -0,0 +1,233 @@
+    if (!field.has_value()) {
+      return InvalidArgument("Field not found in schema: {}", idx);
+    }
+    auto field_name = field.value().get().name();
+    auto view_of_column = array_view.children[idx];
+
+#define PARSE_PRIMITIVE_FIELD(field_name, type)                           \
+  for (size_t row_idx = 0; row_idx < view_of_column->length; row_idx++) { \
+    if (!ArrowArrayViewIsNull(view_of_column, row_idx)) {                 \
+      auto value = ArrowArrayViewGetIntUnsafe(view_of_column, row_idx);   \
+      manifest_files[row_idx].field_name = static_cast<type>(value);      \
+    }                                                                     \
+  }
+
+    if (field_name == ManifestFile::kManifestPath.name()) {
+      for (size_t row_idx = 0; row_idx < view_of_column->length; row_idx++) {
+        if (!ArrowArrayViewIsNull(view_of_column, row_idx)) {
+          auto value = ArrowArrayViewGetStringUnsafe(view_of_column, row_idx);
+          std::string path_str(value.data, value.size_bytes);
+          manifest_files[row_idx].manifest_path = path_str;
+        }
+      }
+    } else if (field_name == ManifestFile::kManifestLength.name()) {
+      PARSE_PRIMITIVE_FIELD(manifest_length, int64_t);
+    } else if (field_name == ManifestFile::kPartitionSpecId.name()) {
+      PARSE_PRIMITIVE_FIELD(partition_spec_id, int32_t);
+    } else if (field_name == ManifestFile::kContent.name()) {
+      for (size_t row_idx = 0; row_idx < view_of_column->length; row_idx++) {
+        if (!ArrowArrayViewIsNull(view_of_column, row_idx)) {
+          auto value = ArrowArrayViewGetIntUnsafe(view_of_column, row_idx);
+          manifest_files[row_idx].content = static_cast<ManifestFile::Content>(value);
+        }
+      }
+    } else if (field_name == ManifestFile::kSequenceNumber.name()) {
+      PARSE_PRIMITIVE_FIELD(sequence_number, int64_t);
+    } else if (field_name == ManifestFile::kMinSequenceNumber.name()) {
+      PARSE_PRIMITIVE_FIELD(min_sequence_number, int64_t);
+    } else if (field_name == ManifestFile::kAddedSnapshotId.name()) {
+      PARSE_PRIMITIVE_FIELD(added_snapshot_id, int64_t);
+    } else if (field_name == ManifestFile::kAddedFilesCount.name()) {
+      PARSE_PRIMITIVE_FIELD(added_files_count, int32_t);
+    } else if (field_name == ManifestFile::kExistingFilesCount.name()) {
+      PARSE_PRIMITIVE_FIELD(existing_files_count, int32_t);
+    } else if (field_name == ManifestFile::kDeletedFilesCount.name()) {
+      PARSE_PRIMITIVE_FIELD(deleted_files_count, int32_t);
+    } else if (field_name == ManifestFile::kAddedRowsCount.name()) {
+      PARSE_PRIMITIVE_FIELD(added_rows_count, int64_t);
+    } else if (field_name == ManifestFile::kExistingRowsCount.name()) {
+      PARSE_PRIMITIVE_FIELD(existing_rows_count, int64_t);
+    } else if (field_name == ManifestFile::kDeletedRowsCount.name()) {
+      PARSE_PRIMITIVE_FIELD(deleted_rows_count, int64_t);
+    } else if (field_name == ManifestFile::kPartitions.name()) {
```

Comment:
What about adding a separate function for the logic in this branch?

Reply:
logical to parse std::vector<PartitionFieldSummary> partitions is really, i will make a new function for it.